index.js

```diff
@@ -248,13 +249,15 @@ module.exports = function (filename, opts) {
 
   function writeBlock(blockIndex) {
     if (!blocksToBeWritten.has(blockIndex)) return
+    writingBlockIndex = blockIndex
     const { block, fileOffset } = blocksToBeWritten.get(blockIndex)
     blocksToBeWritten.delete(blockIndex)
-    const drainsBefore = (waitingDrain.get(blockIndex) || []).slice(0)
 
     debug("writing block of size: %d, to offset: %d",
           block.length, blockIndex * blockSize)
     raf.write(blockIndex * blockSize, block, (err) => {
+      const drainsBefore = (waitingDrain.get(blockIndex) || []).slice(0)
+      writingBlockIndex = -1
```

Comment:
This seems correct, but the thing I'm still scratching my head about is, what does `drainsBefore` and `drainsAfter` now mean? I understood that `drainsBefore` was the array of functions **before** the write is done, but now its meaning changed to be the same as `drainsAfter`? Does this meant that `if (drainsBefore.length === drainsAfter.length)` below (line 285) is always true? Seems like this code needs to be updated to reflect these new semantics.

Reply:
That is a good question. The streams could potentially have added something to the "queue", so I believe this is still true. Maybe I can try and clarify that a bit with some comments.